File: ai-job-portal-backend/candidate-service/src/main/java/com/prahlad/aijobportal/candidateservice/education/service/EducationService.java
package com.prahlad.aijobportal.candidateservice.education.service;

import com.prahlad.aijobportal.candidateservice.education.dto.request.EducationRequest;
import com.prahlad.aijobportal.candidateservice.education.dto.response.EducationResponse;

import java.util.List;
import java.util.UUID;

public interface EducationService {

    EducationResponse create(UUID userId, EducationRequest request);

    List<EducationResponse> getAll(UUID userId);

    EducationResponse update(UUID userId, UUID educationId, EducationRequest request);

    void delete(UUID userId, UUID educationId);
}

File: ai-job-portal-frontend/src/features/profile/services/education.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type { EducationRequest, EducationResponse } from "@/features/profile/types";

/** Maps 1:1 to candidate-service EducationController (/candidate/education). */
export const educationService = {
  getAll: () => apiClient.get<ApiResponse<EducationResponse[]>>("/candidate/education").then((res) => res.data),

  create: (payload: EducationRequest) =>
    apiClient.post<ApiResponse<EducationResponse>>("/candidate/education", payload).then((res) => res.data),

  update: (educationId: string, payload: EducationRequest) =>
    apiClient
      .put<ApiResponse<EducationResponse>>(`/candidate/education/${educationId}`, payload)
      .then((res) => res.data),

  delete: (educationId: string) =>
    apiClient.delete<ApiResponse<null>>(`/candidate/education/${educationId}`).then((res) => res.data),
};
